### PR TITLE
Added support to run DM with IPv4 or IPv6 or both. 

### PR DESCRIPTION
**Pull request by grandsuri**
Added support to run DM with IPv4 or IPv6 or both. On the server side, the binding is done to all the IPs with AF_INET6. On the client side, the socket is created depending on the IP Address family of the given ip address.

### CODE REVIEW

**Comment by grandsuri** [comment]
Validated on VM cluster. Need to try on physical cluster

**Review by nsarras — CHANGES_REQUESTED** [review]
Please test on physical cluster before merging.

File: dss_datamover/socket_communication.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def bind(self, host, port):
             reuse previous sockets which were bound on the same address and remained in TIME_WAIT state.
             """
             self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            self.socket.bind((host, port))
+            self.socket.bind(('', port))
```
[the remark this answers is not in the document]
Comment (grandsuri) [reply dss_datamover/socket_communication.py:266]: Time to refresh some socket stuff from your side

File: dss_datamover/socket_communication.py [hunk 2]
```diff
@@ -76,7 +66,21 @@ def connect(self, host, port):
...
+            ip_info = ipaddress.ip_address(host)
+            if type(ip_info) == ipaddress.IPv4Address:
+                self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            elif type(ip_info) == ipaddress.IPv6Address:
+                self.socket = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+            else:
+                self.logger.error(f"Invalid IP address {host}")
+                raise ConnectionError("Socket initialization failed!")
+        except:
+            self.logger.error("Wrong ip_address_family - {}, Supported {}".format(host, IP_ADDRESS_FAMILY))
```
[the remark this answers is not in the document]
Comment (grandsuri) [reply dss_datamover/socket_communication.py:79]: All the time for the invalid IP, exception will be raised.
But I just added else loop for any unknown case.

**Comment by velomatt** [comment]
I am testing this artifact on VM cluster (IPV4 management, IPV6 MinIO), will updated my results to this thread

**Pull request closed without merging**